hibike/unit_tests/hibike_message_tests.py

```diff
@@ -0,0 +1,385 @@
+"""
+Unit tests for functions in hibike_message.
+"""
+
+
+import unittest
+import random
+import queue
+import time
+import threading
+
+import serial
+
+import hibike_message
+import spawn_virtual_devices
+DEVICE_TYPES = [dev_id for dev_id in hibike_message.DEVICES]
+
+
+def test_with_random_data(func, arg_func, kwarg_func=lambda: {}, times=5, *args, **kwargs):
+    """
+    Test FUNC with random arguments generated by ARG_FUNC and KWARG
+    func, repeating TIMES.
+    """
+    for _ in range(times):
+        # Do not expand list arguments out
+        generated_args = arg_func()
+        if isinstance(generated_args, tuple):
```

Comment:
Yeah, I think you're right about this, especially since the data we are generating should always be of the same type.